modules/auth/proc-adding-gitlab-oauth-as-an-authentication-provider.adoc

```diff
@@ -0,0 +1,103 @@
+For local development::: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+For a non-local deployment::: `pass:[http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/gitlab/handler/frame]`
++
+NOTE: Due to a peculiarity with GitLab OAuth, ensure the URL has no trailing `/` after 'frame'.
+*Scopes*:: Select the following scopes from the list, and validate:
++
+// NOTE: including here verbatim copy from Gitlab screen.
+`read_user`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`read_repository`::: Grants read-only access to the authenticated user's profile through the /user API endpoint, which includes username, public email, and full name.
+Also grants access to read-only API endpoints under /users.
+`write_repository`::: Grants read/write access to repositories on private projects using Git-over-HTTP (not using the API).
+`openid`::: Grants permission to authenticate with GitLab using OpenID Connect.
+Also gives read-only access to the user's profile and group memberships.
+`profile`::: Grants read-only access to the user's profile data using OpenID Connect.
+`email`::: Grants read-only access to the user's primary email address using OpenID Connect.
+
+Save the *Application ID* and *Secret* for the next step.
+
+. Set up the Gitlab authentication provider in the {product} backend (`app-config.yaml`).
++
+Add the `gitlab` provider configuration to your `app-config.yaml` under the root `auth` configuration:
++
+[source,yaml]
+----
+auth:
+  environment: development
+  providers:
+    gitlab:
+      development:
+        clientId: ${AUTH_GITLAB_CLIENT_ID}
+        clientSecret: ${AUTH_GITLAB_CLIENT_SECRET}
+        ## uncomment if using self-hosted GitLab
+        # audience: https://gitlab.company.com
+        ## uncomment if using a custom redirect URI
+        # callbackUrl: https://${BASE_URL}/api/auth/gitlab/handler/frame
+----
++
+The GitLab provider is a structure with these configuration keys:
++
+`clientId`:: The *Application ID* that you generated on GitLab, such as
+`4928c033ab3d592845c044a653bc20583baf84f2e67b954c6fdb32a532ab76c9`.
+`clientSecret`:: The *Secret* that you generated on Gitlab, such as `gloas-f2c9c350759cc08346fbf94a476ae83c579c76dd629fc5eeef9dc21eedfe0475`.
+`audience` (optional):: The base URL for the self-hosted GitLab instance, such as
+`pass:[https://gitlab.company.com]`.
+`callbackUrl` (optional):: The URL matching the *Redirect URI* registered when creating your GitLab OAuth App, such as:
+* For local development: `pass:[http://localhost:7007/api/auth/gitlab/handler/frame]`
+* For a non-local deployment: `pass:[http://{APP_FQDN}:{APP_BACKEND_PORT}/api/auth/gitlab/handler/frame]`
++
+NOTE: Due to a peculiarity with GitLab OAuth, ensure the URL has no trailing `/` after 'frame'.
+
+. (Optionally) To enable the Gitlab provider for sign-in,
```

Comment:
https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/78/files#diff-ac84c51230e4999462622d1e6a25e1a32c3f1992a365b7a560ac6b439091d8fbR64

 

The optional step is currently not supported in RHDH because this change needs engineering effort. and there are some changes needed to be in product to support this optional changes.

 

 

For the community version this documentation is completely supported.